server/data/data-cassandra/src/test/java/org/apache/james/dlp/eventsourcing/cassandra/CassandraEventSourcingDLPConfigurationStoreTest.java

```diff
@@ -20,8 +20,14 @@
 package org.apache.james.dlp.eventsourcing.cassandra;
 
 import org.apache.james.dlp.api.DLPConfigurationStoreContract;
+import org.apache.james.junit.categories.Unstable;
+import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.extension.ExtendWith;
-
+/*
+Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=<resolving>, imagePullPolicy=DefaultPullPolicy())
+Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not build image: unable to find image "sha256:0122a9f1514ea667d656281fb4892bf0f7ce894ed45cfd8250b7cfbc9efacd4d"
```

Comment:
Actually this likely means all testcontainers based tests are not stable on the Apache CI.

Do you confirm?

Maybe we could be finding a way to tern them all off at once, while waiting of a stable solution for running them on Apache INFRA?

Reply:
I don't know, only cassandra related tests have failed this time round and I am very surprised by the image name: 
 "sha256:0122a9f1514ea667d656281fb4892bf0f7ce894ed45cfd8250b7cfbc9efacd4d

is this some kind of snapshot cassandra image ?

Reply:
We rely on a custom, less resource hungry image than the default one: (DockerCassandra)

```
        cassandraContainer = new GenericContainer<>(
            new ImageFromDockerfile(imageName,doNotDeleteImageAfterUsage)
                .withDockerfileFromBuilder(builder ->
                    additionalSteps.applyStep(builder
                        .from("cassandra:3.11.3")
                        .env("ENV CASSANDRA_CONFIG", "/etc/cassandra")
                        .run("echo \"-Xms" + CASSANDRA_MEMORY + "M\" >> " + JVM_OPTIONS)
                        .run("echo \"-Xmx" + CASSANDRA_MEMORY + "M\" >> " + JVM_OPTIONS)
                        .run("sed", "-i", "s/auto_snapshot: true/auto_snapshot: false/g", "/etc/cassandra/cassandra.yaml")
                        .run("echo 'authenticator: PasswordAuthenticator' >> /etc/cassandra/cassandra.yaml")
                        .run("echo 'authorizer: org.apache.cassandra.auth.CassandraAuthorizer' >> /etc/cassandra/cassandra.yaml"))
                        .build()))
            .withTmpFs(ImmutableMap.of("/var/lib/cassandra", "rw,noexec,nosuid,size=200m"))
            .withExposedPorts(CASSANDRA_PORT)
            .withLogConsumer(DockerCassandra::displayDockerLog);
```

Reply:
At this stage, I don't really care the reason, I'll tag everything that fails unstable to reach a reproductible build which isn't the case for now.
Note that tests tagged unstable still run in the next stage which I have never been able to reach at this point because new failures keep popping up all over the place all the time ! 
Once I have separated stable reproductible tests from unstable tests we can all take a bit of time to understand the root cause of each failure by analysing the logs and fixing the root cause. 
If the root cause is really an issue in the CI plateform, we will be able to open an INFRA  ticket with qualified information so that they can fix it, at which point we will be able to untag all the corresponding tests